Assets/Scripts/SpellScripts/Fireball.cs

```diff
@@ -40,7 +42,20 @@ public void OnTriggerEnter2D(Collider2D otherObject)
         // Destroy the fireball object if it collides with anything except the player or the grid
         if (!otherObject.gameObject.CompareTag("Player") && !otherObject.gameObject.CompareTag("Grid"))
         {
-            StartCoroutine(WaitForExplosionAnimationDestroyFireball());
```

Comment:
I think it was better to leave the StartCoroutine method here, because the fireballs would get destroyed when out of map bounds.